### DISCUSSION TITLE
Reducer Nested Object Update

### DISCUSSION THREAD

**Discussion by jd-carroll** [reduxjs/redux-toolkit#4367] [Q&A] [answered]
I am reasonably certain I know the answer to this, but just thought I would check...

I _want to_ write a reducer that contains the following action:

```typescript
export const MySlice = createSlice({
  name: 'sliced',
  initialState,
  reducers: {
    recordValue: (state: SliceState, action: PayloadAction<{ foo: Bar }>) => {
      state.parent = state.parent || {};
      state.parent.foo = action.payload.foo;
    }
  }
})
```

But something is telling me that is wrong and that the action creator should be:
```typescript
    recordValue: (state: SliceState, action: PayloadAction<{ foo: Bar }>) => {
      state.parent = { ...state.parent };
      state.parent.foo = action.payload.foo;
    }
```

Or maybe it doesn't matter because Immer is able to detect the change any ways?

If I added two selectors:
```typescript
  selectors: {
    SelectParent: (state: SliceState) => state.parent,
    SelectFoo: (state: SliceState) => state.parent.foo
  }
```

Is there a difference when the selectors are triggered for updates?

**Comment by markerikson** [comment, marked as answer]
It looks like that `state.parent =` line is just filling in a default value, right?

Personally, I'd write it as:

```ts
if (!state.parent) {
  state.parent = {}
}
```

(Or if you really want to get fancy, the "nullish coalescing assignment operator": `state.parent ??= {}`)

**Reply by EskiMojo14** [reply]
`(state.parent ??= {}).foo = action.payload.foo`
😎
